crates/factor-outbound-networking/src/runtime_config/spin.rs

```diff
@@ -0,0 +1,273 @@
+use anyhow::{bail, ensure, Context};
+use serde::{Deserialize, Deserializer};
+use spin_factors::runtime_config::toml::GetTomlValue;
+use std::io;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use super::{validate_host, TlsConfig};
+
+pub struct SpinTlsRuntimeConfig {
+    runtime_config_dir: PathBuf,
+}
+
+impl SpinTlsRuntimeConfig {
+    pub fn new(runtime_config_dir: impl Into<PathBuf>) -> Self {
```

Comment:
Let's document `runtime_config_dir`. This is the directory that that relative paths in the runtime config toml will be resolved against, correct? Perhaps we can name a document that so this is clear without needing to review the code.